webapp/packages/core-blocks/src/useSync.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * CloudBeaver - Cloud Database Manager
+ * Copyright (C) 2020-2025 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0.
+ * you may not use this file except in compliance with the License.
+ */
+
+import { mutex } from '@dbeaver/js-helpers';
+import { useEffect, useState } from 'react';
+import { useObjectRef } from './useObjectRef.js';
+
+interface ISyncHook {
+  markOutdated(): void;
+}
+
+export function useSync(callback: () => void | Promise<void>): ISyncHook {
```

Comment:
The name of this hook is too generic. Could we give it a more explicit name so it’s clearer why we need it?

Reply:
I agree; it's generic (and it's really generic), but I can't imagine another name. This hook aims to provide a simple way to implement a React-based reaction to side effects. Generally, we have imperative code that will invoke the function right away. This leads to problems when an event might be called multiple times, so we will have numerous functions running in concurrent mode.

That leads to problems like this one you fixed earlier: https://github.com/dbeaver/cloudbeaver/pull/3734

`useSync` "connects" events with React lifecycle, so the interface can react to changes in a React manner by utilizing React's built-in batching functionality.